modules/core/CameraPerspective.h

```diff
@@ -28,13 +28,24 @@ namespace rt {
 		 * @param up Normalized camera up-vector
 		 * @param angle (Vertical) full opening angle of the viewing frustum in degrees
 		 */
-		DllExport CCameraPerspective(Size resolution, const Vec3f& pos, const Vec3f& dir, const Vec3f& up, float angle)
+
+		DllExport CCameraPerspective(Size resolution, const Vec3f& pos, const Vec3f& dir,
+			const Vec3f& up, float angle)
 			: ICamera(resolution)
 			, m_pos(pos)
 			, m_dir(dir)
 			, m_up(up)
 			, m_focus(1.0f / tanf(angle * Pif / 360))	// f = 1 / tg(angle / 2)
 		{}
+
+		DllExport CCameraPerspective(Size resolution, const Vec3f& pos, const Vec3f& dir, 
```

Comment:
I just think if it will make more sense to put the implementation in a separate class... For example in a
`CCameraThinLens : public CCameraPerspective`. What do you think about it? What would be the advantage to keep the implementation here ?

Reply:
CCameraThinLens class would be nice, but the only problem I see is that we would need two additional classes: ` CCameraThinLensPerspective : public CCameraPerspective` and `CCameraThinLensOrthographic : public CCameraOrthographic` since we have DoF for both cameras. 

If we are going to use single class maybe we can derive it from both like this: `CCameraThinLens:  public CCameraPerspective, public CCameraOrthographic`. But then we need a flag in the constructor for `CCameraThinLens` to check if it's orthographic or perspective.

Reply:
And  `CCameraPerspective` and `CCameraOrthographic` are pinhole cameras, so we can think of them as an idealization of a thinlense camera as the aperture shrinks to zero. So maybe `CCameraThinLens` is more general which is why I added lense radius, focal distance properties to the main Camera class

Reply:
or maybe we can add `enable_dof(radius, focal distance,...)` method in ` CCameraOrthographic`  and `CCameraPerspective`

Reply:
Yes, you are right - to keep the architecture consistent it makes sense to have the DoF code in one place and to make it possible to extend any camera (perspective and orthographic, with target versions) to have DoF effect... Creating the Camera***ThinLense looks like a natural way to extend the current architecture, however we should not double the code. The idea of adding one method `enable_dof()` also deserves attention. However I keen more to the idea of derived classes at the moment.
I suggest for the moment being add only 1 class `CCameraThinLens : public CCameraPerspective`. I believe that after that point and after this pull request, we will have a better understanding how to extend the rest of the cameras to use DoF.

Reply:
sure, will implement CCameraThinLens